pkg/tsdb/postgres/postgres.go

```diff
@@ -42,16 +47,40 @@ func newPostgresQueryEndpoint(datasource *models.DataSource) (tsdb.TsdbQueryEndp
 	return sqleng.NewSqlQueryEndpoint(&config, &queryResultTransformer, newPostgresMacroEngine(timescaledb), logger)
 }
 
-func generateConnectionString(datasource *models.DataSource) string {
+func generateConnectionString(datasource *models.DataSource, logger log.Logger) (string, error) {
 	sslmode := datasource.JsonData.Get("sslmode").MustString("verify-full")
+
+	// Always pass SSL mode
+	sslopts := "sslmode=" + url.QueryEscape(sslmode)
+
+	// Attach root certificate if provided
+	if sslrootcert := datasource.JsonData.Get("sslrootcertfile").MustString(""); sslrootcert != "" {
+		logger.Debug("Setting CA certificate", "sslrootcert", sslrootcert)
+		sslopts += "&sslrootcert=" + url.QueryEscape(sslrootcert)
+	}
+
+	// Attach client certificate and key if both are provided
+	sslcert := datasource.JsonData.Get("sslcertfile").MustString("")
+	sslkey := datasource.JsonData.Get("sslkeyfile").MustString("")
```

Comment:
```suggestion
	sslkey := datasource.JsonData.Get("ssl_keyfile").MustString("")
```

Reply:
Why underscore?

Reply:
@marefr Torkel asked for it

Reply:
To be specific he asked for snake case in JsonData fields, but he didn't specify exactly where to insert the underscores. Looking it over again, I see the underscore here is maybe inconsistent with e.g. "sslroot_certfile". What do you think would be best?

Reply:
Weird. We use camel case for all datasource settings, see https://grafana.com/docs/grafana/latest/administration/provisioning/#example-datasource-config-file

Reply:
To conclude - it turned out Torkel actually wanted camel case :) I've fixed up my suggestions to use camel case for JsonData fields.